### PR TITLE
feat: add block_id for serialized block

### LINKED ISSUE #291: Should return block_id when getting block by flavour

### PR DESCRIPTION
**Pull request by thorseraq**
close #291 

## Before

```json
{
      "sys:children": [],
      "sys:created": 1679473264652.0,
      "sys:flavor": "list",
      "sys:version": [
          1.0,
          0.0
      ]
  }
```

## After
```json
{
      "sys:id": "0",
      "sys:children": [],
      "sys:created": 1679473264652.0,
      "sys:flavor": "list",
      "sys:version": [
          1.0,
          0.0
      ]
  }
```

### CODE REVIEW

**Comment by darkskygit** [comment]
image
This behavior does not conform to the standard, we need to discuss it later, in the standard we use `sys:id` to represent the block id, and do not include the space id

**Comment by thorseraq** [comment]
> image
> 
> This behavior does not conform to the standard, we need to discuss it later, in the standard we use `sys:id` to represent the block id, and do not include the space id

Sorry, I didn't notice this. How about using `sys:id` in replace of `block_id`

```rust
{
    "sys:id": "0",
    "sys:children": [],
    "sys:created": 1679473264652.0,
    "sys:flavor": "list",
    "sys:version": [
        1.0,
        0.0
    ]
}
```

**Comment by darkskygit** [comment]
> > image
> > This behavior does not conform to the standard, we need to discuss it later, in the standard we use `sys:id` to represent the block id, and do not include the space id
> 
> Sorry, I didn't notice this. How about using `sys:id` in replace of `block_id`
> 
> ```rust
> {
>     "sys:id": "0",
>     "sys:children": [],
>     "sys:created": 1679473264652.0,
>     "sys:flavor": "list",
>     "sys:version": [
>         1.0,
>         0.0
>     ]
> }
> ```

bingo

File: libs/jwst/src/block.rs [hunk 1]
```diff
@@ -535,7 +536,16 @@ impl Serialize for Block {
...
         let trx = self.doc.transact();
         let any = self.block.to_json(&trx);
-        any.serialize(serializer)
+
+        let mut buffer = String::new();
+        any.to_json(&mut buffer);
+        let any: Value = serde_json::from_str(&buffer).unwrap();
+
+        let mut block = any.as_object().unwrap().clone();
+        block.insert("block_id".to_string(), Value::String(self.block_id.clone()));
```
Comment (darkskygit) [inline libs/jwst/src/block.rs:545]: sys:id

File: libs/jwst/src/block.rs [hunk 2]
```diff
@@ -535,7 +536,16 @@ impl Serialize for Block {
...
+        block.insert("space_id".to_string(), Value::String(self.space_id.clone()));
```
Comment (darkskygit) [inline libs/jwst/src/block.rs:546]: not need

**Review by darkskygit — CHANGES_REQUESTED** [review]

File: libs/jwst/src/block.rs (hunk 1 above)
Comment (thorseraq) [reply libs/jwst/src/block.rs]: updated

File: libs/jwst/src/block.rs (hunk 2 above)
Comment (thorseraq) [reply libs/jwst/src/block.rs]: updated

**Review by darkskygit — APPROVED** [review]

**Pull request merged**